ui/components/component-library/text-field/text-field.test.js

```diff
@@ -38,118 +44,84 @@ describe('TextField', () => {
-        clearButtonIconProps={{ 'data-testid': 'clear-button-icon' }}
-        rightAccessory={<div>right-accessory</div>}
-        showClear
-      />,
-    );
-    const textField = getByRole('textbox');
-    fireEvent.change(textField, { target: { value: 'text value' } });
-    expect(textField.value).toBe('text value');
-    fireEvent.click(getByTestId('clear-button'));
-    expect(textField.value).toBe('');
-  });
-  it('should fire onClear event when passed to onClear prop', () => {
-    const onClear = jest.fn();
-    const { getByRole, getByTestId } = render(
+  it('should fire onClick event when passed to clearButtonProps when clear button is clicked', async () => {
+    // As showClearButton is intended to only work with a controlled input we need to pass a value and onChange prop
+    const fn = jest.fn();
+    const { user, getByRole } = setup(
       <TextField
-        onClear={onClear}
-        clearButtonProps={{ 'data-testid': 'clear-button' }}
-        clearButtonIconProps={{ 'data-testid': 'clear-button-icon' }}
-        showClear
+        value="text"
+        onChange={() => {}}
+        clearButtonOnClick={fn}
+        showClearButton
       />,
     );
-    const textField = getByRole('textbox');
-    fireEvent.change(textField, { target: { value: 'text value' } });
-    expect(textField.value).toBe('text value');
-    fireEvent.click(getByTestId('clear-button'));
-    expect(onClear).toHaveBeenCalledTimes(1);
+    await user.click(getByRole('button', { name: /Clear/u }));
+    expect(fn).toHaveBeenCalledTimes(1); // clear button onClick is fired
   });
-  it('should fire clearButtonProps.onClick event when passed to clearButtonProps.onClick prop', () => {
-    const onClear = jest.fn();
-    const onClick = jest.fn();
-    const { getByRole, getByTestId } = render(
+  it('should fire clearButtonProps.onClick event when passed to clearButtonProps.onClick prop', async () => {
+    // As showClearButton is intended to only work with a controlled input we need to pass a value and onChange prop
+    const fn = jest.fn();
+    const { user, getByRole } = setup(
       <TextField
-        onClear={onClear}
-        clearButtonProps={{ 'data-testid': 'clear-button', onClick }}
-        clearButtonIconProps={{ 'data-testid': 'clear-button-icon' }}
-        showClear
+        value="text"
+        onChange={() => {}}
+        clearButtonProps={{ onClick: fn }}
+        showClearButton
       />,
     );
-    const textField = getByRole('textbox');
-    fireEvent.change(textField, { target: { value: 'text value' } });
-    expect(textField.value).toBe('text value');
-    fireEvent.click(getByTestId('clear-button'));
-    expect(onClear).toHaveBeenCalledTimes(1);
-    expect(onClick).toHaveBeenCalledTimes(1);
+    await user.click(getByRole('button', { name: /Clear/u }));
+    expect(fn).toHaveBeenCalledTimes(1);
   });
   it('should be able to accept inputProps', () => {
     const { getByRole } = render(
```

Comment:
When running these tests locally you may run into this propType issue it's a known issue and I have an open PR that with the gracious help of @Gudahtt we are still trying to work through https://github.com/MetaMask/metamask-extension/pull/16078

<img width="547" alt="Screen Shot 2022-11-09 at 7 33 04 PM" src="https://user-images.githubusercontent.com/8112138/200979200-95e94f84-f20c-4f08-a6ca-9f1033870b86.png">